### ISSUE TITLE
Flex changes not loading correctly

### ISSUE THREAD

**Issue by shahzeb79** [SAP/ui5-tooling#245]
Hi Colleagues,
                      We face a problem with 50+ teams who uses flex changes in there tests. And while running tests through karma or tooling, these changes are not loaded successfully and wrong controls are displayed in tests, hence tests fails. So it will be really good if i can get more details on how we can make flex changes work for normal freestyle fiori app setup.

Thanks
Shahzeb

**Comment by matz3** [comment]
Could you please provide some insights on your observations and more details about this? As this problem is not known to me and there is no example on how to reproduce we can't provide any help right now.
Also information about the UI5 version would be great as I suspect this rather to be a regression within UI5 (e.g. sap.ui.fl).

**Comment by RandomByte** [comment]
I had a quick chat with Shahzeb earlier today. I suspected that this might be because we only have flex related tasks, and no middleware. Or should flex changes already work from the sources?

**Comment by RandomByte** [comment]
Hi @shahzeb79,

I just discussed this with @Lonwyr from the UI5 Flexibility area. This is a known gap and there are a couple of ways to close it. Most likely on the UI5 Tooling side, with a new ui5-server middleware that serves a list of available changes files for a project (=namespace) and a connector to consume that list.

We agreed to follow up on this and let you know of any new developments. Thanks for reporting!

**Comment by Lonwyr** [comment]
**TLDR**
- Flex changes are only applied if they are in the bundles generated out of the build
- If you need to run the sources without the minified / preload (i.e. to check code coverage), you can do this with the sap-ui-debug=true parameter
- Loading them directly without a build step is only possible in the IDEs (SAP Business Application Studio, SAP WebIDE - as of today)

**details for the interested community**
We load on an app startup the flex files from the generated bundles. Since we are part of the build which is highly recommended form every IDE allowing the creation of changes, we know they are in the preload if they exist. Therefore, we do not send any request if they are NOT preloaded.
This is overruled in case the preload is not considered (i.e. in debug).
If no build step run, concrete implementations on the IDE side provide the data by registering their own data provider. Otherwise, OpenUI5 has no clue where to find the single files since they have a GUID in their names and listing a namespace is dependent on the server serving the app. We are coming closer and closer to a solution and we will discuss how we can provide a satisfying solution for this known gap - at least for the most used server(s) on which you may run your apps for testing.
